src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.cs

```diff
@@ -596,6 +596,7 @@ internal void FinishOperationSyncFailure(SocketError socketError, int bytesTrans
             // the attempt socket will be closed anyways, so not updating the state is OK.
             // If we're doing a static ConnectAsync to an IPEndPoint, we need to dispose
             // of the socket, as we manufactured it and the caller has no opportunity to do so.
+            SocketType? socketType = _currentSocket?.SocketType;
```

Comment:
The assignment of this variable should be moved into the if block below and should use `currentSocket` rather than `_currentSocket`.

Reply:
Sorry, I seem to be missing the point, could you please clarify? I see there is a check for null [on line 601](https://github.com/dotnet/runtime/blob/d9fade2bfa0e00111945e9e4bf15705129dc4b38/src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.cs#L601) which implies `_currentSocket` can be not null here. Furthermore, `currentSocket` is assigned only on the next line 600 and nowhere else which means it will get the same value as `_currentSocket`.

Reply:
Transferred bytes logging in failure cases has been removed completely per suggestion below.